Assets/MixedRealityToolkit.SDK/Inspectors/Input/Handlers/BaseInputHandlerInspector.cs

```diff
@@ -23,5 +23,32 @@ public override void OnInspectorGUI()
             EditorGUILayout.PropertyField(isFocusRequiredProperty);
             serializedObject.ApplyModifiedProperties();
         }
+
+        /// <summary>
+        /// If MRTK is not initialized in scene, then show an error message and button to add MRTK to the scene
+        /// If MRTK does not has an active profile, show an error message
+        /// </summary>
+        /// <returns>true if MRTK is initialized and has active profile, false otherwise</returns>
+        protected bool CheckMixedRealityToolkit()
+        {
+            if (!MixedRealityToolkit.IsInitialized)
```

Comment:
A MixedRealityToolkit, instance is no longer required to use the MRTK (this is in the process of becoming easier to configure).

This method should likely not be added.

Reply:
@davidkline-ms if there is no MRTK instance in scene, how does one get the list of available inputactions? Looking at the updated Interactable.cs, it still checks if MRTK is initialized. 

       ```
 public static bool TryGetInputActions(out string[] descriptionsArray)
        {
            if (!MixedRealityToolkit.ConfirmInitialized() || !MixedRealityToolkit.Instance.HasActiveProfile)
            {
                descriptionsArray = null;
                return false;
            }

            MixedRealityInputAction[] actions = InputSystem.InputSystemProfile.InputActionsProfile.InputActions;

            descriptionsArray = new string[actions.Length];
            for (int i = 0; i < actions.Length; i++)
            {
                descriptionsArray[i] = actions[i].Description;
            }

            return true;
        }
```

Reply:
The code snippet above uses a cached input system that was acquired using

```
MixedRealityServiceRegistry.TryGetService<IMixedRealityInputSystem>()
```

This pattern is supported for the MixedRealityToolkit object as well as standalone services.

Please let me know if you need more clarity or have questions.

Reply:
@davidkline-ms , yes it uses a cached input system. The point though is that the input actions list is only grabbed from the MRTK instance currently in scene. If there is no MRTK instance in scene, then you can't get the input actions list. Thus, my CheckMixedRealityToolkit() is still necessary. There is still a dependency

Reply:
The primary concern is to not add any new references to MixedRealityToolkit.Instance. Let's discuss and post the results here.